examples/plugins/ebpf/main.go

```diff
@@ -0,0 +1,179 @@
+package main
+
+import (
+	"bufio"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"net"
+	"net/http"
+	"os"
+	"os/exec"
+	"sync"
+	"time"
+)
+
+func main() {
+	hostname, _ := os.Hostname()
+	var (
+		addr   = flag.String("addr", "/var/run/scope/plugins/ebpf.sock", "unix socket to listen for connections on")
+		hostID = flag.String("hostname", hostname, "hostname of the host running this plugin")
+	)
+	flag.Parse()
+
+	log.Println("Starting...")
+
+	// Compile and run the ebpf script
+	ebpf := exec.Command("http-parse-simple.py")
+	ebpf.Stderr = os.Stderr
+	stdout, err := ebpf.StdoutPipe()
+	if err != nil {
+		log.Fatal(err)
+	}
+	if err := ebpf.Start(); err != nil {
+		log.Fatal(err)
+	}
+	defer func() {
+		if ebpf.Process != nil {
+			ebpf.Process.Kill()
+		}
+	}()
+
+	os.Remove(*addr)
```

Comment:
> What if another instance is already running?

sneaky replacement happens. Convenient for dev, but probably best not left in.

> It would be helpful to have some helpers taking care of this stuff (socket creation registration, automatic handshake handle etc ..).

Yep. Pulling the socket management out into a couple helper functions would make it more obvious when/how you should do it.